chia/data_layer/download_data.py

```diff
@@ -1,166 +1,141 @@
-            }
-            await ws.send_str(json.dumps(request))
-            msg = await ws.receive()
-            root_json = json.loads(msg.data)
-            generation = root_json["generation"]
-            root = await data_store.get_tree_root(tree_id=tree_id, lock=lock)
-            # We've downloaded too much, rollback.
-            if root.generation > generation:
-                await data_store.rollback_to_generation(tree_id=tree_id, target_generation=generation, lock=lock)
-                return True
-            existing_generation = root.generation + 1
-            while existing_generation <= generation:
-                request = {
-                    "type": "request_operations",
-                    "tree_id": tree_id.hex(),
-                    "generation": str(existing_generation),
-                    "max_generation": str(generation),
-                }
-                await ws.send_str(json.dumps(request))
-                msg = await ws.receive()
-                msg_json = json.loads(msg.data)
-
-                for row in msg_json:
-                    if row["is_insert"]:
-                        await data_store.insert(
-                            bytes.fromhex(row["key"]),
-                            bytes.fromhex(row["value"]),
-                            tree_id,
-                            None
-                            if row["reference_node_hash"] == "None"
-                            else (bytes32.from_hexstr(row["reference_node_hash"])),
-                            None if row["side"] == "None" else Side(row["side"]),
-                            status=Status(row["root_status"]),
-                            lock=lock,
-                        )
-                    else:
-                        await data_store.delete(
-                            bytes.fromhex(row["key"]),
-                            tree_id,
-                            status=Status(row["root_status"]),
-                            lock=lock,
-                        )
-                    current_root = await data_store.get_tree_root(tree_id, lock=lock)
-                    if current_root.node_hash is None:
-                        if row["hash"] != "None":
-                            return False
-                    else:
-                        if current_root.node_hash.hex() != row["hash"]:
-                            return False
-                    existing_generation += 1
-
-            await ws.send_str(json.dumps({"type": "close"}))
+    for root_hash in root_hashes:
+        existing_generation += 1
+        filename = get_delta_filename(tree_id, root_hash, existing_generation)
+        url = f"http://{ip}:{port}/{filename}"
 
-    return True
+        try:
+            async with aiohttp.ClientSession() as session:
+                async with session.get(url) as resp:
+                    if resp.status != 200:
+                        raise RuntimeError("Didn't get 200 response status.")
```

Comment:
https://docs.aiohttp.org/en/stable/client_reference.html?highlight=raise%20for%20status#aiohttp.ClientResponse.raise_for_status

```suggestion
                    resp.raise_for_status()
```

I think you could alternatively just do `aiohttp.ClientSession(raise_for_status=True)` above.